apps/examples/fs_performance/fs_performance_main.c

```diff
@@ -0,0 +1,502 @@
+					close(frt_fd);
+					close(fd);
+					return ERROR;
+				}
+			}
+
+			close(fd);
+			ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after);
+			ioctl(frt_fd, TCIOC_STOP, 0);
+			close(frt_fd);
+
+			tt2 = after.timeleft - before.timeleft;
+			tt += tt2;
+			printf(" iteration : %d, time : %lu\n", i, tt2);
+		}
+		printf("total time :  %lu us\n", tt);
+
+		ret = unlink(file[0]);
+		if (ret != OK) {
+			printf("Failed to unlink file %s, ret : %d, errno : %d\n", file[0], ret, errno);
+			return ERROR;
+		}
+
+		break;
+
+	case TEST_FLAG_APPEND :
+		memset(rw_buf, 0, BUFFER_SIZE);
+                tt = tt2 = 0;
+		for(int it = 0; it <= 5; it++) {
+			fd = open(file[0], O_WROK | O_CREAT);
+			if (fd < 0) {
+				printf("Unable to open file : %s, fd = %d\n", file[0], fd);
+				return ERROR;
+			}
+			/*create a test file of size f_sz */
+			ret = write(fd, rw_buf, f_sz);
+			if (ret != (f_sz)) {
+				printf("Unable to write in file: %s, size = %d, ret = %d\n", file[0], f_sz, ret);
+				close(fd);
+				return ERROR;
+			}
+                	close(fd);
+
+			frt_fd = open("/dev/timer0", O_RDONLY);
+			if (frt_fd < 0) {
+                                printf("Unable to do open file : /dev/timer0, errno : %d\n", file[0], errno);
+                                return ERROR;
+                        }
+			ioctl(frt_fd, TCIOC_SETMODE, MODE_FREERUN);
+			ioctl(frt_fd, TCIOC_START, TRUE);
+			ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&before);
+
+			fd = open(file[0], O_WROK | O_APPEND);
+			if (fd < 0) {
+				printf("Unable to open file : %s, ret = %d\n", file[0], fd);
+				close(frt_fd);
+				return ERROR;
+			}
+
+			/*append 64 kb data*/
+			for (j = 0; j < (64 * 1024) / b_sz; j++ ) {
+				ret = write(fd, rw_buf, b_sz);
+				if (ret != (b_sz)) {
+					printf("Unable to append to file: %s, ret = %d\n", file[0], ret);
+					close(frt_fd);
+					close(fd);
+					return ERROR;
+				}
+			}
+			close(fd);
+
+			ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after);
+			ioctl(frt_fd, TCIOC_STOP, 0);
+			close(frt_fd);
+
+			tt2 = after.timeleft - before.timeleft;
+			tt += tt2;
+			printf("iteration : %d,time : %lu\n", it, tt2);
+
+			ret = unlink(file[0]);
+			if (ret != OK) {
+				printf("Failed to unlink file %s, ret : %d, errno : %d\n", file[0], ret, errno);
+				return ERROR;
+			}
+		}
+		printf("total time : %lu us\n", tt);
+		break;
+
+	case TEST_FLAG_READ :
+		memset(rw_buf, 0, BUFFER_SIZE);
+		fd = open(file[0], O_WROK | O_CREAT);
+                if (fd < 0) {
+			printf("Unable to open file : %s,  ret : %d\n", file[0], fd);
+                        return ERROR;
+                }
+
+                ret = write(fd, rw_buf, 512);
+                if (ret != (512)) {
+			printf("Unable to write in file: %s, ret : %d\n", file[0], ret);
+			close(fd);
+			return ERROR;
+                }
+		close(fd);
+		for (int file_size = 512; file_size <= (f_sz); file_size *= 2) {
```

Comment:
Integer definitions like these I think we do at the start of the function as per TizenRT coding standards
Not sure if that applies to apps as well, can you check and confirm please?

Reply:
Some have used like these before :
https://github.com/Samsung/TizenRT/pull/4698/files#diff-f8d3e8d376542d4a01703bea6c7e1bdf119c7d149b2ca6d64953f233ea5d1e7dR79

Reply:
Let's have the 2 loop iterators declared at the start of this function also declared in the loop statement itself.
I checked other apps like powercut, they have the iterator declared in the loop itself, so that okay.